js_src/rdm.js

```diff
@@ -5,146 +5,973 @@ angular.module('rdmApp', [])
+   };
+
+  $scope.RDM = RDM;
+  $scope.command_classes = [];
+  angular.forEach(RDM.COMMAND_CLASS, function(value, key) {
+    this.push({
+      'value': value,
+      'label': key + ' (' + formatService.toHex(value, 2, '0x') + ')'
+     });
+   }, $scope.command_classes);
+  $scope.command_classes.sort(function(a, b) {
+   return (( a.value === b.value ) ? 0 : ( ( a.value > b.value ) ? 1 : -1 ));
+  });
+
+  $scope.OUTPUT_FORMAT = {};
+  angular.forEach(OUTPUT_FORMAT, function(value, key) {
+    this[value.label] = value.value;
+   }, $scope.OUTPUT_FORMAT);
+
+  $scope.output = '';
+  $scope.error = '';
 
-   var manufacturer0 = (manufacturer >> 8) & 0xff;
-   var manufacturer1 = manufacturer & 0xff;
-   var device0 = (device >> 24) & 0xff;
-   var device1 = (device >> 16) & 0xff;
-   var device2 = (device >> 8) & 0xff;
-   var device3 = device & 0xff;
-
-   euid_bytes.push(manufacturer0 | 0xaa);
-   euid_bytes.push(manufacturer0 | 0x55);
-   euid_bytes.push(manufacturer1 | 0xaa);
-   euid_bytes.push(manufacturer1 | 0x55);
-
-   euid_bytes.push(device0 | 0xaa);
-   euid_bytes.push(device0 | 0x55);
-   euid_bytes.push(device1 | 0xaa);
-   euid_bytes.push(device1 | 0x55);
-   euid_bytes.push(device2 | 0xaa);
-   euid_bytes.push(device2 | 0x55);
-   euid_bytes.push(device3 | 0xaa);
-   euid_bytes.push(device3 | 0x55);
-
-   var checksum = euid_bytes.reduce(
-    function(previousValue, currentValue, index, array) {
-     return previousValue + currentValue;
-    }
-   );
-   var checksum0 = (checksum >> 8) & 0xff;
-   var checksum1 = checksum & 0xff;
-
-   euid_bytes.push(checksum0 | 0xaa);
-   euid_bytes.push(checksum0 | 0x55);
-   euid_bytes.push(checksum1 | 0xaa);
-   euid_bytes.push(checksum1 | 0x55);
-
-   convertor.euid = euid_bytes.map(function(i) {
-    return i.toString(16);
-   }).join(' ');
+  var parseUIDOrSetError = function(uid, uid_name) {
+   var parse_return = parserService.uidToBytes(uid);
+   if (parse_return[0]) {
+    $scope.error = 'Invalid ' + uid_name + ' UID:' + parse_return[0];
+    return '';
+   }
+   return parse_return[1];
+  };
+
+  var showHideParamData = function() {
+   $scope.properties.show_param_data = (
+       rdmHelperService.isGetSet($scope.packet.command_class) &&
+       ($scope.packet.response_type === RDM.RESPONSE_TYPE.ACK ||
+        $scope.packet.response_type === RDM.RESPONSE_TYPE.ACK_OVERFLOW));
   };
+
+  var commandClassChanged = function(new_value, old_value, scope) {
+    $scope.packet.command_class = new_value;
+    $scope.properties.is_request = rdmHelperService.isRequest(new_value);
+    showHideParamData();
+
+    var pid_names = [];
+    if (new_value === RDM.COMMAND_CLASS.DISCOVERY_COMMAND) {
+     pid_names = ['DISC_UNIQUE_BRANCH', 'DISC_MUTE', 'DISC_UN_MUTE'];
+    } else if (new_value === RDM.COMMAND_CLASS.DISCOVERY_COMMAND_RESPONSE) {
+     pid_names = ['DISC_MUTE', 'DISC_UN_MUTE'];
+    } else {
+     var discovery_pids = ['DISC_UNIQUE_BRANCH', 'DISC_MUTE', 'DISC_UN_MUTE'];
```

Comment:
I guess the future JSON API can return suitable lists based on supported command classes.